CSC8503/SuspicionSystem/LocationBasedSuspicion.h

```diff
@@ -2,11 +2,61 @@
 #include "SuspicionMetre.h"
 #include <cmath>
 #include <vector>
+#include "../NCLCoreClasses/Vector3.h"
+#include "../NCLCoreClasses/Vector2.h"
+#include <limits>
+
+using namespace NCL::Maths;
 
 namespace SuspicionSystem
 {
     const float MAX_NEARBY_DISTANCE = 5;
-    const float DT_UNTIL_LOCATION_RECOVERY = 5;
+    struct CantorPair{
+
+        CantorPair(){
+            mValue = 0;
+        }
+
+        CantorPair(Vector3 inPos){
+            Vector3 shiftedPos;
+            shiftedPos.x = int(int(inPos.x) > 0 ? int(inPos.x) * 2 : int(inPos.x) * -2 + 1);
+            shiftedPos.z = int(int(inPos.z) > 0 ? int(inPos.z) * 2 : int(inPos.z) * -2 + 1);
+
+            mValue = (shiftedPos.x + shiftedPos.z) * 
+                     (shiftedPos.x + shiftedPos.z + 1) / 2 + shiftedPos.z;
+        };
+
+        bool operator<(const CantorPair& other) const {
+            return mValue < other.mValue;
+        }   
+
+        int operator*() const {
+            return mValue;
+        }
+
+        operator int() const {
+            return mValue;
+        }
+
+        CantorPair& operator=(int value) {
+            mValue = value;
+            return *this;
+        }
+        
+        static Vector3 InverseCantorPair(CantorPair inPair){
+            int w = int((sqrt(8 * inPair + 1) - 1) / 2);
+            int t = (w * w + w) / 2;
+            Vector3 shiftedPos;
+            shiftedPos.z = inPair - t;
+            shiftedPos.x = w - shiftedPos.z;
+            Vector3 outPos;
+            outPos.x = (int(shiftedPos.x) % 2 == 0 ? shiftedPos.x / 2 : -(shiftedPos.x / 2) );
+            outPos.z = (int(shiftedPos.z) % 2 == 0 ? shiftedPos.z / 2 : -(shiftedPos.z / 2) );
+            return outPos;
```

Comment:
If we can have some comments on this struct and it's functions it'll be very useful. Because at first glance I couldn't understand the math in this function and I don't know why we used canton pairs.

Reply:
And renaming variables with more meaningful and understandable names can be very useful as well.